### ISSUE TITLE
I2V First&Last Frame and Video continuation

### ISSUE THREAD

**Issue by kaibioinfo** [Wan-Video/Wan2.1#283]
Hi, in the Wan paper you write

> In practice, the aforementioned approach as illustrated in Fig. 18 is extensible to other controllable
generation tasks, such as first-last frame transformation and video continuation. The proposed
masking mechanism explicitly delineates the input conditions and specifies the frames to be generated.
To optimize the efficacy of the masking strategy, we incorporate multiple tasks, i.e., image-to-video
generation, video continuation, first-last frame transformation, and random frame interpolation, into
our unified framework. Specifically, we adopt the following training paradigm for the I2V model. During the joint training phase, we conduct unified pre-training of our mask-guided model across
these diverse tasks. This phase facilitates the model’s ability to discern which positions should be
preserved and which should be generated, depending on the input content. Subsequently, in the
fine-tuning stage, we concentrate on refining the model’s performance on each individual task.

For me this sounds like the I2V model can also be used to inpaint videos or to position the conditioning image (or multiple images) on any position.

However, I never succeeded in trying this. I wanted to generate a looped video, so I placed the conditioning image in the first and last frame and specified the mask accordingly. But the generated video was severely damaged/full of artifacts. My assumption was that the temporal component of the VAE does not allow to place a latent video frame chunk on any position. So I generated a static video consisting only of a single image repeated along the time axis and encoded that via the vae. I then set the mask on the first and last frame of the video and did i2v. Again, the outcome was severely damaged and full of artifacts.

**Is there a way to implement looped videos and/or video inpaints with the I2V model?** I know, you are also working on VACE that will allow such stuff and many more. But it should already be possible with the current I2V model, shouldn't it?

**Comment by danielzyy1990** [comment]
Right now the released I2V model only supports basic i2v func, which dont follow the unified  paradigm mentioned in the paper. We will release a model supports First&Last frames this week.

**Comment by 571129857** [comment]
> Right now the released I2V model only supports basic i2v func, which dont follow the unified paradigm mentioned in the paper. We will release a model supports First&Last frames this week.

hi, I have the same problem as upstairs, looking forward to the support of the first and last frames, thank you

**Comment by leioukupo** [comment]
> Right now the released I2V model only supports basic i2v func, which dont follow the unified paradigm mentioned in the paper. We will release a model supports First&Last frames this week.

where is  the released model that  supports First&Last frames

**Comment by LiuXiaolong19920720** [comment]
> Right now the released I2V model only supports basic i2v func, which dont follow the unified paradigm mentioned in the paper. We will release a model supports First&Last frames this week.

Any update about this?
